druid-shell/src/backend/wayland/surfaces/surface.rs

```diff
@@ -376,12 +379,22 @@ impl Data {
             for rect in damaged_region.rects() {
                 // Convert it to physical coordinate space.
                 let rect = buffers::RawRect::from(*rect).scale(self.scale.get());
-                self.wl_surface.borrow().damage_buffer(
-                    rect.x0,
-                    rect.y0,
-                    rect.x1 - rect.x0,
-                    rect.y1 - rect.y0,
-                );
+
+                if self.wl_surface.borrow().as_ref().version()
+                    >= wl_surface::REQ_DAMAGE_BUFFER_SINCE
+                {
+                    self.wl_surface.borrow().damage_buffer(
+                        rect.x0,
+                        rect.y0,
+                        rect.x1 - rect.x0,
+                        rect.y1 - rect.y0,
+                    );
+                } else {
+                    // We don't care about obscure pre version 4 compositors
+                    // and just damage the whole surface instead of
+                    // translating from buffer coordinates to surface coordinates
+                    self.wl_surface.borrow().damage(0, 0, i32::MAX, i32::MAX);
```

Comment:
druid actually uses the damaged region in order to optimize its own drawing, so I think if you damage the whole surface here then you'll end up with a partially rendered window. Which is probably no big deal if all real compositors support this. But I think it at least deserves a warning...

Reply:
Would `self.invalidate()` do the job? The same way we do it for `force == true`

Reply:
Yep, I guess that should take care of telling druid that it needs to draw everything. I'm a little confused about that `invalidate()` though, because shouldn't we also be calling `self.wl_surface.borrow().damage(...)` in that branch?

Reply:
I think it should be called, as far as I can tell what happens when no damage is sent is not specified in the protocol. So compositor may just do nothing in that case.

Reply:
Done